src/node_contextify.h

```diff
@@ -106,6 +106,9 @@ class ContextifyContext : public BaseObject {
       const v8::ScriptCompiler::Source& source);
   static void ContainsModuleSyntax(
       const v8::FunctionCallbackInfo<v8::Value>& args);
+  static void ShouldRetryAsESM(const v8::FunctionCallbackInfo<v8::Value>& args);
+  static bool ShouldRetryAsESMInternal(Environment* env,
```

Comment:
Actually, none of the functions between `CompileFunctionAndCacheResult` and this belong to `ContextifyContext` because they have nothing to do with `ContextifyContext`.....but that was already added by the detection-module PR.

Reply:
Okay . . . so in some future PR you want to split these into a new class? That’s fine.

I’m not sure why you say they have nothing to do with `ContextifyContext`. The new functions and `CompileFunction` are all compiling source via V8, so they seem very related/similar.

Reply:
`ContextifyContext` is the C++ equivalent of `vm.Context`. `vm.compileFunction` interacts with it because it takes an optional parsing context (that's not the main context). Other stuff below it doesn't interact with vm context in any way, they are all doing things in the main context.